src/transformers/training_args.py

```diff
@@ -679,13 +679,17 @@ def __post_init__(self):
             self.hf_deepspeed_config.trainer_config_process(self)
 
     def __repr__(self):
-        # We override the default repr to remove deprecated arguments from the repr. This method should be removed once
-        # those deprecated arguments are removed form TrainingArguments. (TODO: v5)
         self_as_dict = asdict(self)
+
+        # We remove deprecated arguments from the repr. That code should be removed once
+        # those deprecated arguments are removed from TrainingArguments. (TODO: v5)
         del self_as_dict["per_gpu_train_batch_size"]
         del self_as_dict["per_gpu_eval_batch_size"]
-        attrs_as_str = [f"{k}={v}" for k, v in self_as_dict.items()]
-        return f"{self.__class__.__name__}({', '.join(attrs_as_str)})"
+
+        attrs_as_str = [f"{k}={v},\n" for k, v in sorted(self_as_dict.items())]
+        return f"{self.__class__.__name__}(\n{''.join(attrs_as_str)})"
+
+    __str__ = __repr__
```

Comment:
Isn't this automatically done by Python? Just wondering.

Reply:
It should be, especially since in the super-class they are the same:

```
python -c "import dataclasses; print(dataclasses.__str__, dataclasses.__repr__)"
<method-wrapper '__str__' of module object at 0x7f660b29a900> <method-wrapper '__repr__' of module object at 0x7f660b29a900>
```

But if I remove it, nothing happens - i.e. the custom `__repr__` doesn't get called.

Reading through the docs in general it's recommended that `__str__` should be implemented and `__repr__` to be overridden with `__str__`.

I will study this some more.

Reply:
Ok, just wanted to double-check it was necessary!

Reply:
How about we use `__str__` directly, since this is really the functionality what we are after.

Reply:
No it's nice to have both return the same thing.

Reply:
sure, I aliased the two.

Let me know if you want repr to have the body and str be the alias instead.

Reply:
I'm fine either way :-)
Thanks for putting back the alias.